ftest/src/test/java/org/jboss/arquillian/graphene/ftest/enricher/TestGrapheneElement.java

```diff
@@ -116,7 +113,7 @@ public void testOneWithJavascript() {
 
     @Test
     public void testOneWithActions() {
-        actions.moveToElement(root).doubleClick().build().perform();
+        new Actions(browser).moveToElement(root).doubleClick().build().perform();
```

Comment:
Correct. Selenium reworked Actions API and it no longer needed, or can be injectable.
Since this is a major version update, this would be acceptable.
Existing code that relies on Actinos API would be completely broken anyway because of what Selenium did, so this would not be an issue.

Reply:
Selenium changed `Actions` class pretty drastically in 4.x. It used to be an interface and now it's a concrete class without any interfaces that it implements. I don't think Arquillian can inject a concrete class, at least not that I could make work.

I don't think this is a problem in since we are upgrading to a new major version for all artifacts.

Reply:
Thank you for the explanation. @mabartos do you know of any test suite which could rely on it? I wonder how big of an impact this removal is.

Reply:
Your comment inspired me to look at this again. With my new deepener understanding how Graphene actually works, I was able to restore the injection as it was before.